File: doc-examples/hibernate-reactive-example-java/src/main/java/example/Manufacturer.java

package example;

import io.micronaut.configuration.hibernate.jpa.proxy.GenerateProxy;
import org.hibernate.annotations.BatchSize;

import jakarta.persistence.*;

@Entity
@GenerateProxy
@BatchSize(size = 10)
public class Manufacturer {
    @Id
    @GeneratedValue
    private Long id;
    private String name;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: doc-examples/hibernate-example-python/src/main/python/example/Manufacturer.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.persistence import Entity, GeneratedValue, Id
from micronaut.configuration.hibernate.jpa.proxy import GenerateProxy
from org.hibernate.annotations import BatchSize


@Entity
@GenerateProxy
@BatchSize(size=10)
@dataclass
class Manufacturer:
    name: str | None = None
    id: Annotated[int | None, Id, GeneratedValue] = None
